x/gov/migrations/v046/migrate.go

```diff
@@ -0,0 +1,97 @@
+package v046
+
+import (
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	"github.com/cosmos/cosmos-sdk/x/gov/types/v1beta2"
+)
+
+// ConvertToLegacyProposal takes a new proposal and attempts to convert it to the
+// legacy proposal format. This conversion is best effort. New proposal types that
+// don't have a legacy message will return a "nil" content.
+func ConvertToLegacyProposal(proposal v1beta2.Proposal) (v1beta1.Proposal, error) {
+	legacyProposal := v1beta1.Proposal{
+		ProposalId:       proposal.ProposalId,
+		Content:          nil,
```

Comment:
```suggestion
```

Reply:
I added this here to make it a bit more explicit to the reader what we were doing but I can remove it

Reply:
No strong opinion, feel free to keep it.